sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveMetastoreCatalogSuite.scala

```diff
@@ -50,6 +51,88 @@ class HiveMetastoreCatalogSuite extends SparkFunSuite with Logging {
     logInfo(df.queryExecution.toString)
     df.as('a).join(df.as('b), $"a.key" === $"b.key")
   }
+
+}
+
+class ParquetLocationSelectionSuite extends DataSourceTest {
+  override def _sqlContext: SQLContext = TestHive
+  val hmc = new HiveMetastoreCatalog(null, TestHive)
+  // ensuring temp directories
+  val baseDir = {
+    val base = File.createTempFile("selectParquetLocationDirectories", "1", hiveFilesTemp)
+    base.delete()
+    base.mkdirs()
+    base
+  }
+
+  test(s"With Selector selecting from ${baseDir.toString}") {
+    val fullpath = (somewhere: String, sometable: String)
+    => s"${baseDir.toString}/${somewhere}/$sometable"
+
+    TestHive.setHadoopFileSelector(new HadoopFileSelector() {
+      override def getFilesSizeInBytes(tableName: String, fs: FileSystem, basePath: Path):
+      Option[Long] = Some(100L)
+
+      override def selectFiles(sometable: String, fs: FileSystem, somewhere: Path):
+      Option[Seq[Path]] =
+        Some(Seq(new Path(fullpath(somewhere.toString, sometable))))
+    })
+
+    // ensure directory existence for somewhere/sometable
+    val somewhereSometable = new File(fullpath("somewhere", "sometable"))
+    somewhereSometable.mkdirs()
+    // somewhere/sometable is a directory => will be selected
+    assertResult(Seq(fullpath("somewhere", "sometable")))(
```

Comment:
nit: preference in the codebase for `assertResult` that extends over more than one line appears to be:
```scala
assertResult(foo) {
  .
  .
  .
}
```